adding scroll view and two timer screens

shrevin opened the pull request.

Commit:
adding scroll view and two timer screens

arthpatel29 commented on Splash/Splash.xcworkspace/xcuserdata/shreyavinjamuri.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:
```diff
@@ -685,8 +685,8 @@
             filePath = "Splash/Networking/SpotifyAPIManager.m"
```
This file should not be part of commit. I think we need to add this to ignore file if not already. 

If there is no ignore file, just look up git ignore file and how to add it in the repo.

arthpatel29 replied:
I think we need to add this to gitignore file as discussed in the previous commit. This is local file, we do not have to commit this.

Commit:
Fixing Scroll View Bug

This commit is to implement feedback from the prior pr and make the scroll view show up in the details screen.

shrevin commented:
[](https://drive.google.com/file/d/1XaFlfi_rmkKxcdjJk7C7WNMgEbv-STxb/view?usp=sharing)

arthpatel29 commented on Splash/.gitignore:
```diff
@@ -0,0 +1 @@
+SpotifyAPIManager.m
```
Is this intended. This would mean any changes to SpotifyAPIManager would not be included in the commit. 
Example for gitignore file: https://github.com/github/gitignore/blob/main/Objective-C.gitignore

yaxuanwang commented on Splash/Splash/Description/DescriptionViewController.swift:
```diff
@@ -0,0 +1,59 @@
+class DescriptionViewController: UIViewController {
+    let redView = UIView(frame: CGRect(x: 20, y: 100, width: 140, height: 100))
+    let animation = CABasicAnimation()
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        redView.backgroundColor = .systemRed
+        view.addSubview(redView)
+        // Do any additional setup after loading the view.
+        animateRight();
+        animateDown()
+    }
```
Are the animation visible?How about moving your animation from `viewDidLoad` to `viewDidAppear`. Usually you are not able to animate the components of your view until it has finished laying out.

shrevin replied:
Hi Yaxuan, the animation does show up. I tried putting it in viewDidAppear, however I don't want the animation to happen every time you go back to the screen if you've already opened it.

arthpatel29 commented on Splash/Splash/Description/DescriptionViewController.swift:
```diff
@@ -0,0 +1,59 @@
+        redView.layer.position = CGPoint(x: 300, y: 600)
+    }
+    
+//    override func viewDidAppear(_ animated: Bool) {
+//        super.viewDidAppear(animated)
+//        let gradientLayer = CAGradientLayer()
+//        gradientLayer.frame = view.bounds
+//        gradientLayer.colors = [UIColor.blue.cgColor,  UIColor.purple.cgColor]
+//        view.layer.addSublayer(gradientLayer)
+//
+//    }
```
Let's make sure commented out code is not committed. You can keep it locally. While creating commit just do not include it. I think you can do this using github desktop app.

yaxuanwang commented on Splash/Splash/TimeWithMusic/TimerXIBView.m:
```diff
@@ -0,0 +1,214 @@
+    [[SpotifyAPIManager shared].appRemote.playerAPI resume:nil];
+}
+
+//- (void)applicationWillResignActive:(UIApplication *)application {
+//    if ([SpotifyAPIManager shared].appRemote.isConnected) {
+//        [[SpotifyAPIManager shared].appRemote disconnect];
+//    }
+//}
+//
+//- (void)applicationDidBecomeActive:(UIApplication *)application {
+//    if ([SpotifyAPIManager shared].appRemote.connectionParameters.accessToken) {
+//        [[SpotifyAPIManager shared].appRemote connect];
+//    }
+//}
```
Not needed?

yaxuanwang commented on Splash/Splash/TimeWithoutMusic/NoMusicViewController.m:
```diff
@@ -33,26 +32,16 @@ @implementation NoMusicViewController
-    dateFormatNoMusic.timeZone = [NSTimeZone timeZoneWithAbbreviation:@"PST"];
-    self.user = [PFUser currentUser];
-    // layout vc
+    
```
nit: You don't need this method if just calling `[super viewDidLoad];`

Commit:
Implementing feedback and creating Description View Controller

created a core animation for descriptions to pop up for each stat

Commit:
Creating functionality to see friends

Used the SWTableViewCell library to implement swipeable buttons the the right

shrevin commented:
For the latest commit:

![IMG_3879](https://user-images.githubusercontent.com/36344124/181614552-f3fb97d2-5153-4962-9b27-4e57c2b65781.PNG)

arthpatel29 commented on Splash/Splash/TimeWithMusic/TimerXIBView.h:
```diff
@@ -13,6 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @interface TimerXIBView : UIView
 - (void) updateTime;
 - (void) updateFontSize:(int)size;
+@property UIViewController *root;
```
This will create a retain cycle when accessed from inside NoMusicViewController. Add 'weak' annotation to make it a weak reference.

arthpatel29 commented on Splash/Splash/TimeWithoutMusic/NoMusicViewController.m:
```diff
@@ -29,6 +29,10 @@ @implementation NoMusicViewController
 bool isPausedNoMusic;
 
+- (void) viewDidLoad {
+    self.stopwatch.root = self;
```
As mentioned above, this will create a retain cycle.

arthpatel29 commented on Splash/Podfile:
```diff
@@ -8,6 +8,7 @@ target 'Splash' do
   # Pods for Splash
   pod 'Parse'
   pod 'Parse/UI'
+  pod 'SWTableViewCell', '~> 0.3.7'
```
I think we can get swipe buttons natively in UITableView, you don't have to use a library for that. Search for 'UITableViewRowAction'

arthpatel29 commented on Splash/Splash/Friends/ViewFriendsViewController.m:
```diff
@@ -0,0 +1,100 @@
+    self.tableView.dataSource = self;
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+    // Do any additional setup after loading the view.        
+        //_testArray = [[NSMutableArray alloc] init];
```
Let make sure we remove commented out code from commit.

arthpatel29 commented on Splash/.gitignore:
```diff
@@ -7,23 +7,7 @@ xcuserdata/
-!default.mode2v3
-*.perspectivev3
-!default.perspectivev3
+Breakpoints_v2.xcbkptlist
```
Not sure if this will work as intended. 
Giving another reference: https://medium.com/codex/create-git-ignore-file-for-swift-and-xcode-development-d7faaf58893c

Commit:
configuring gitignore

Commit:
fixing gitignore

Commit:
Fixing bug with Spotify and voice countdown

Added code to restart connection when voice countdown is played

Commit:
removing bool as a pointer
